src/crewai/cli/tools/main.py

```diff
@@ -103,23 +103,40 @@ def install(self, handle: str):
             )
             raise SystemExit
 
-        self._add_repository_to_poetry(get_response.json())
         self._add_package(get_response.json())
 
         console.print(f"Succesfully installed {handle}", style="bold green")
 
-    def _add_repository_to_poetry(self, tool_details):
-        repository_handle = f"crewai-{tool_details['repository']['handle']}"
-        repository_url = tool_details["repository"]["url"]
-        repository_credentials = tool_details["repository"]["credentials"]
+    def login(self):
+        login_response = self.plus_api_client.login_to_tool_repository()
+
+        if login_response.status_code != 200:
+            console.print(
+                "Failed to authenticate to the tool repository. Make sure you have the access to tools.",
+                style="bold red",
+            )
+            raise SystemExit
+
+        login_response_json = login_response.json()
```

Comment:
One of the issues we had in yesterday's incident is related to the fact the response was not a json object. Should we consider this here and somehow print the body? cc @pythonbyte.

Reply:
I created a method that checks if the response is OK, otherwise it logs the error. Let me know what you think: 61b786254bdbfd3f421649d79dbf0a4fa4657982

Reply:
Lovely! ❤️